### PR TITLE
Implement support for the terraform-provider-mikrotik

### PR DESCRIPTION
**Pull request by ddelnano**
This adds support for the [mikrotik terraform provider](https://github.com/ddelnano/terraform-provider-mikrotik) I develop. This provider has support for dhcp leases, however, in practice these are not something you would track when they are ephemeral. However, I do want to track the leases that are static since those are IPs that I never want to change.

Importing these on a one off basis has been tedious and so my terraform code doesn't stay up to date. However, this project makes it easy to keep this up to date with a single command! I will likely add more resource support if this is accepted but for now this was the resource that was giving me the most pain managing.

## Testing
- [x] Ran the following commands and verified the output files look correct.
```
$  ./terraformer import mikrotik -r dhcp_lease --filter='Name=dynamic;Value=false'
2020/05/01 09:55:03 mikrotik importing... dhcp_lease
2020/05/01 09:55:03 [INFO] Running the mikrotik command: `[/ip/dhcp-server/lease/print]`
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--emby-002D--002A-E
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer---002A-4
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer---002A-9
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--archiso-002D--002A-A19
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--prometheus-002D--002A-5
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--puppet-002D--002A-D
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--devbox-002D--002A-E05
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--radarr-002D--002A-6
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--nas-002D--002A-E66
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--raspberrypi-002D--002A-A
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer---002A-B
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--xoa-002D--002A-2
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--sensu-002D--002A-1
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--jira-002D--002A-3
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--switchab525d-002D--002A-7
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--raspberrypi-002D--002A-12FC
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--confluence-002D--002A-1695
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--HS110-002D--002A-176C
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--k8s-002D-master-002D--002A-1A2F
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--SonosZP-002D--002A-2C92
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--Google-002D-Home-002D-Mini-002D--002A-2EC6
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer---002A-2FAC
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer---002A-30CF
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--Google-002D-Home-002D-Mini-002D--002A-32A9
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--amazon-002D-d277725ce-002D--002A-32AF
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--ddelnansmacbook-002D--002A-34B7
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--Gaurav-002D--002A-34BC
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--DakotaTusIphone-002D--002A-34F3
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--User2s-002D-MBP-002D--002A-34F4
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--DomsiPhone11pro-002D--002A-34F7
2020/05/01 09:55:03 Refreshing state... mikrotik_dhcp_lease.tfer--Yelp-002D-s-002D-iPad-002D--002A-3521
2020/05/01 09:55:04 mikrotik Connecting....
2020/05/01 09:55:04 mikrotik save dhcp_lease
2020/05/01 09:55:04 mikrotik save tfstate for dhcp_lease

$ cat generated/mikrotik/dhcp_lease/dhcp_lease.tf
resource "mikrotik_dhcp_lease" "tfer---002A-30CF" {
  address    = "192.168.88.78"
  comment    = "Unifi AP AC Pro 2"
  dynamic    = "false"
  macaddress = "E0:63:DA:3C:45:F1"
}

resource "mikrotik_dhcp_lease" "tfer---002A-4" {
  address    = "192.168.88.88"
  dynamic    = "false"
  macaddress = "4C:D9:8F:37:E1:C6"
}

resource "mikrotik_dhcp_lease" "tfer---002A-9" {
  address    = "192.168.88.253"
  comment    = "Unifi AP AC Pro 1"
  dynamic    = "false"
  macaddress = "78:8A:20:5C:68:D7"
}

resource "mikrotik_dhcp_lease" "tfer---002A-B" {
  address    = "192.168.88.89"
  dynamic    = "false"
  macaddress = "4C:D9:8F:37:E1:C5"
}

resource "mikrotik_dhcp_lease" "tfer--HS110-002D--002A-176C" {
  address    = "192.168.88.151"
  dynamic    = "false"
  hostname   = "HS110"
  macaddress = "68:FF:7B:B8:80:0E"
}

resource "mikrotik_dhcp_lease" "tfer--archiso-002D--002A-A19" {
  address    = "192.168.88.254"
  dynamic    = "false"
  hostname   = "archiso"
  macaddress = "74:D4:35:5B:78:CD"
}

resource "mikrotik_dhcp_lease" "tfer--confluence-002D--002A-1695" {
  address    = "192.168.88.161"
  dynamic    = "false"
  hostname   = "confluence"
  macaddress = "FE:58:E0:BB:75:E2"
}

resource "mikrotik_dhcp_lease" "tfer--devbox-002D--002A-E05" {
  address    = "192.168.88.108"
  dynamic    = "false"
  hostname   = "devbox"
  macaddress = "E6:79:0A:3F:5B:02"
}

resource "mikrotik_dhcp_lease" "tfer--emby-002D--002A-E" {
  address    = "192.168.88.41"
  dynamic    = "false"
  hostname   = "emby"
  macaddress = "3A:97:C1:C6:91:E4"
}

resource "mikrotik_dhcp_lease" "tfer--jira-002D--002A-3" {
  address    = "192.168.88.80"
  dynamic    = "false"
  hostname   = "jira"
  macaddress = "D6:5E:D0:09:DC:8D"
}

resource "mikrotik_dhcp_lease" "tfer--k8s-002D-master-002D--002A-1A2F" {
  address    = "192.168.88.194"
  dynamic    = "false"
  hostname   = "k8s-master"
  macaddress = "F6:BA:A9:5C:E8:16"
}

resource "mikrotik_dhcp_lease" "tfer--nas-002D--002A-E66" {
  address    = "192.168.88.112"
  dynamic    = "false"
  hostname   = "nas"
  macaddress = "5A:CF:23:80:0D:EA"
}

resource "mikrotik_dhcp_lease" "tfer--prometheus-002D--002A-5" {
  address    = "192.168.88.79"
  dynamic    = "false"
  hostname   = "prometheus"
  macaddress = "B2:5E:51:5A:0D:A0"
}

resource "mikrotik_dhcp_lease" "tfer--puppet-002D--002A-D" {
  address    = "192.168.88.22"
  dynamic    = "false"
  hostname   = "puppet"
  macaddress = "DE:AB:9A:6E:44:8D"
}

resource "mikrotik_dhcp_lease" "tfer--radarr-002D--002A-6" {
  address    = "192.168.88.32"
  dynamic    = "false"
  hostname   = "radarr"
  macaddress = "66:1F:5D:28:F4:9D"
}

resource "mikrotik_dhcp_lease" "tfer--raspberrypi-002D--002A-12FC" {
  address    = "192.168.88.137"
  dynamic    = "false"
  hostname   = "raspberrypi"
  macaddress = "DC:A6:32:20:49:99"
}

resource "mikrotik_dhcp_lease" "tfer--raspberrypi-002D--002A-A" {
  address    = "192.168.88.10"
  dynamic    = "false"
  hostname   = "raspberrypi"
  macaddress = "B8:27:EB:3E:FB:6C"
}

resource "mikrotik_dhcp_lease" "tfer--sensu-002D--002A-1" {
  address    = "192.168.88.27"
  dynamic    = "false"
  hostname   = "sensu"
  macaddress = "06:ED:B7:94:5B:87"
}

resource "mikrotik_dhcp_lease" "tfer--switchab525d-002D--002A-7" {
  address    = "192.168.88.90"
  dynamic    = "false"
  hostname   = "switchab525d"
  macaddress = "08:CC:A7:AB:52:5D"
}

resource "mikrotik_dhcp_lease" "tfer--xoa-002D--002A-2" {
  address    = "192.168.88.86"
  dynamic    = "false"
  hostname   = "xoa"
  macaddress = "D6:D3:8A:2D:9A:4F"
}

$  ./terraformer import mikrotik -r dhcp_lease
2020/05/01 10:01:15 mikrotik importing... dhcp_lease
2020/05/01 10:01:15 [INFO] Running the mikrotik command: `[/ip/dhcp-server/lease/print]`
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--radarr-002D--002A-6
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--switchab525d-002D--002A-7
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer---002A-9
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--raspberrypi-002D--002A-A
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--jira-002D--002A-3
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--sensu-002D--002A-1
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer---002A-4
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--emby-002D--002A-E
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer---002A-B
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--puppet-002D--002A-D
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--xoa-002D--002A-2
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--devbox-002D--002A-E05
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--archiso-002D--002A-A19
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--prometheus-002D--002A-5
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--nas-002D--002A-E66
2020/05/01 10:01:15 Refreshing state... mikrotik_dhcp_lease.tfer--raspberrypi-002D--002A-12FC
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--confluence-002D--002A-1695
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--HS110-002D--002A-176C
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--k8s-002D-master-002D--002A-1A2F
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--SonosZP-002D--002A-2C92
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--Google-002D-Home-002D-Mini-002D--002A-2EC6
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer---002A-2FAC
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer---002A-30CF
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--Google-002D-Home-002D-Mini-002D--002A-32A9
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--amazon-002D-d277725ce-002D--002A-32AF
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--ddelnansmacbook-002D--002A-34B7
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--DakotaTusIphone-002D--002A-34F3
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--Gaurav-002D--002A-34BC
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--User2s-002D-MBP-002D--002A-34F4
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--DomsiPhone11pro-002D--002A-34F7
2020/05/01 10:01:16 Refreshing state... mikrotik_dhcp_lease.tfer--Yelp-002D-s-002D-iPad-002D--002A-3521
2020/05/01 10:01:16 mikrotik Connecting....
2020/05/01 10:01:16 mikrotik save dhcp_lease
2020/05/01 10:01:16 mikrotik save tfstate for dhcp_lease

$ cat generated/mikrotik/dhcp_lease/dhcp_lease.tf
resource "mikrotik_dhcp_lease" "tfer---002A-2FAC" {
  address    = "192.168.88.48"
  dynamic    = "true"
  macaddress = "E8:61:7E:8E:F1:81"
}

resource "mikrotik_dhcp_lease" "tfer---002A-30CF" {
  address    = "192.168.88.78"
  comment    = "Unifi AP AC Pro 2"
  dynamic    = "false"
  macaddress = "E0:63:DA:3C:45:F1"
}

resource "mikrotik_dhcp_lease" "tfer---002A-4" {
  address    = "192.168.88.88"
  dynamic    = "false"
  macaddress = "4C:D9:8F:37:E1:C6"
}

resource "mikrotik_dhcp_lease" "tfer---002A-9" {
  address    = "192.168.88.253"
  comment    = "Unifi AP AC Pro 1"
  dynamic    = "false"
  macaddress = "78:8A:20:5C:68:D7"
}

resource "mikrotik_dhcp_lease" "tfer---002A-B" {
  address    = "192.168.88.89"
  dynamic    = "false"
  macaddress = "4C:D9:8F:37:E1:C5"
}

resource "mikrotik_dhcp_lease" "tfer--DakotaTusIphone-002D--002A-34F3" {
  address    = "192.168.88.17"
  dynamic    = "true"
  hostname   = "DakotaTusIphone"
  macaddress = "5C:1D:D9:82:97:81"
}

resource "mikrotik_dhcp_lease" "tfer--DomsiPhone11pro-002D--002A-34F7" {
  address    = "192.168.88.59"
  dynamic    = "true"
  hostname   = "DomsiPhone11pro"
  macaddress = "74:42:8B:87:67:D6"
}

resource "mikrotik_dhcp_lease" "tfer--Gaurav-002D--002A-34BC" {
  address    = "192.168.88.43"
  dynamic    = "true"
  hostname   = "Gaurav"
  macaddress = "40:9C:28:DA:E1:74"
}

resource "mikrotik_dhcp_lease" "tfer--Google-002D-Home-002D-Mini-002D--002A-2EC6" {
  address    = "192.168.88.11"
  dynamic    = "true"
  hostname   = "Google-Home-Mini"
  macaddress = "20:DF:B9:81:9F:54"
}

resource "mikrotik_dhcp_lease" "tfer--Google-002D-Home-002D-Mini-002D--002A-32A9" {
  address    = "192.168.88.54"
  dynamic    = "true"
  hostname   = "Google-Home-Mini"
  macaddress = "00:F6:20:5B:3F:1F"
}

resource "mikrotik_dhcp_lease" "tfer--HS110-002D--002A-176C" {
  address    = "192.168.88.151"
  dynamic    = "false"
  hostname   = "HS110"
  macaddress = "68:FF:7B:B8:80:0E"
}

resource "mikrotik_dhcp_lease" "tfer--SonosZP-002D--002A-2C92" {
  address    = "192.168.88.250"
  dynamic    = "true"
  hostname   = "SonosZP"
  macaddress = "B8:E9:37:B1:19:2C"
}

resource "mikrotik_dhcp_lease" "tfer--User2s-002D-MBP-002D--002A-34F4" {
  address    = "192.168.88.39"
  dynamic    = "true"
  hostname   = "User2s-MBP"
  macaddress = "8C:85:90:35:2D:5B"
}

resource "mikrotik_dhcp_lease" "tfer--Yelp-002D-s-002D-iPad-002D--002A-3521" {
  address    = "192.168.88.62"
  dynamic    = "true"
  hostname   = "Yelp-s-iPad"
  macaddress = "C0:B6:58:B3:21:5A"
}

resource "mikrotik_dhcp_lease" "tfer--amazon-002D-d277725ce-002D--002A-32AF" {
  address    = "192.168.88.248"
  dynamic    = "true"
  hostname   = "amazon-d277725ce"
  macaddress = "18:74:2E:B5:F7:EE"
}

resource "mikrotik_dhcp_lease" "tfer--archiso-002D--002A-A19" {
  address    = "192.168.88.254"
  dynamic    = "false"
  hostname   = "archiso"
  macaddress = "74:D4:35:5B:78:CD"
}

resource "mikrotik_dhcp_lease" "tfer--confluence-002D--002A-1695" {
  address    = "192.168.88.161"
  dynamic    = "false"
  hostname   = "confluence"
  macaddress = "FE:58:E0:BB:75:E2"
}

resource "mikrotik_dhcp_lease" "tfer--ddelnansmacbook-002D--002A-34B7" {
  address    = "192.168.88.19"
  dynamic    = "true"
  hostname   = "ddelnansmacbook"
  macaddress = "78:4F:43:50:5B:A3"
}

resource "mikrotik_dhcp_lease" "tfer--devbox-002D--002A-E05" {
  address    = "192.168.88.108"
  dynamic    = "false"
  hostname   = "devbox"
  macaddress = "E6:79:0A:3F:5B:02"
}

resource "mikrotik_dhcp_lease" "tfer--emby-002D--002A-E" {
  address    = "192.168.88.41"
  dynamic    = "false"
  hostname   = "emby"
  macaddress = "3A:97:C1:C6:91:E4"
}

resource "mikrotik_dhcp_lease" "tfer--jira-002D--002A-3" {
  address    = "192.168.88.80"
  dynamic    = "false"
  hostname   = "jira"
  macaddress = "D6:5E:D0:09:DC:8D"
}

resource "mikrotik_dhcp_lease" "tfer--k8s-002D-master-002D--002A-1A2F" {
  address    = "192.168.88.194"
  dynamic    = "false"
  hostname   = "k8s-master"
  macaddress = "F6:BA:A9:5C:E8:16"
}

resource "mikrotik_dhcp_lease" "tfer--nas-002D--002A-E66" {
  address    = "192.168.88.112"
  dynamic    = "false"
  hostname   = "nas"
  macaddress = "5A:CF:23:80:0D:EA"
}

resource "mikrotik_dhcp_lease" "tfer--prometheus-002D--002A-5" {
  address    = "192.168.88.79"
  dynamic    = "false"
  hostname   = "prometheus"
  macaddress = "B2:5E:51:5A:0D:A0"
}

resource "mikrotik_dhcp_lease" "tfer--puppet-002D--002A-D" {
  address    = "192.168.88.22"
  dynamic    = "false"
  hostname   = "puppet"
  macaddress = "DE:AB:9A:6E:44:8D"
}

resource "mikrotik_dhcp_lease" "tfer--radarr-002D--002A-6" {
  address    = "192.168.88.32"
  dynamic    = "false"
  hostname   = "radarr"
  macaddress = "66:1F:5D:28:F4:9D"
}

resource "mikrotik_dhcp_lease" "tfer--raspberrypi-002D--002A-12FC" {
  address    = "192.168.88.137"
  dynamic    = "false"
  hostname   = "raspberrypi"
  macaddress = "DC:A6:32:20:49:99"
}

resource "mikrotik_dhcp_lease" "tfer--raspberrypi-002D--002A-A" {
  address    = "192.168.88.10"
  dynamic    = "false"
  hostname   = "raspberrypi"
  macaddress = "B8:27:EB:3E:FB:6C"
}

resource "mikrotik_dhcp_lease" "tfer--sensu-002D--002A-1" {
  address    = "192.168.88.27"
  dynamic    = "false"
  hostname   = "sensu"
  macaddress = "06:ED:B7:94:5B:87"
}

resource "mikrotik_dhcp_lease" "tfer--switchab525d-002D--002A-7" {
  address    = "192.168.88.90"
  dynamic    = "false"
  hostname   = "switchab525d"
  macaddress = "08:CC:A7:AB:52:5D"
}

resource "mikrotik_dhcp_lease" "tfer--xoa-002D--002A-2" {
  address    = "192.168.88.86"
  dynamic    = "false"
  hostname   = "xoa"
  macaddress = "D6:D3:8A:2D:9A:4F"
}

```
- [x] Submitted the CLA

## Todo
- [x] Do we need some verification that the vendor dependency changes are ok? I tried to make the minimum set of changes by running the following commands, however, I'm new to how the go toolchain manages the vendor directory now.
```
go get github.com/ddelnano/terraform-provider-mikrotik@0.2.2
go mod vendor
```

### CODE REVIEW

**Comment by sergeylanzman** [comment]
Thanks!
About vendors, please delete vendor files, and rebase to one commit.
I start "unvendoring" process in terraformer. No need push vendors code

File: README.md [hunk 1]
```diff
@@ -1292,6 +1294,25 @@ List of supported [commercetools](https://commercetools.com/de/) resources:
 *   `types`
     * `commercetools_type`
 
+### Use with [Mikrotik](https://commercetools.com/de/)
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply README.md]: Yep this was a copy and paste error, fixed!

**Comment by ddelnano** [comment]
@sergeylanzman I removed the vendor code, fixed the README and rebased to a single commit.

**Comment by meshuga** [comment]
@sergeylanzman That's great to hear! Updating dependencies will be less painful :)
@ddelnano Thanks!

**Pull request merged**